### PR TITLE
Changes to ObjectArg and added IdArg

### LINKED ISSUE #20: Object id additions
At the moment we want to make a `pmac.DlsPmacAsynMotor` have a `pmac.Geobrick` as a `PORT`. But what if we wanted to allow a `pmac.PowerBrick` to be passed as well? iocbuilder allowed us to make a `pmac.DeltaTau` that `pmac.Geobrick` could subclass from, should be allow the same inheritance here?

Once we have decided that, we need to put enough information in the schema to allow the GUI to restrict selection of `PORT` to `pmac.DeltaTau` instances. This is not a supported concept in JSON schema, so we need to invent our own vocab like https://gregsdennis.github.io/json-everything/usage/vocabs-unique-keys.html

### PR DESCRIPTION
**Pull request by niamhdougan**
Fixes #20

### CODE REVIEW

File: src/ibek/ioc.py [hunk 1]
```diff
@@ -67,24 +68,25 @@ def make_entity_class(definition: Definition, support: Support) -> Type[Entity]:
...
         if isinstance(arg, ObjectArg):
 
-            def make_conversion(name: str = arg.type) -> Conversion:
-                module_name, entity_name = name.split(".", maxsplit=1)
-                entity_cls = getattr(getattr(modules, module_name), entity_name)
+            def lookup_instance(id__):
+                try:
+                    return id_to_entity[id__]
+                except KeyError:
+                    raise ValidationError(f"{id__} is not in {list(id_to_entity)}")
```
Comment (thomascobb) [inline src/ibek/ioc.py:79]: We can back out the `id__` name now

```suggestion
            def lookup_instance(id):
                try:
                    return id_to_entity[id]
                except KeyError:
                    raise ValidationError(f"{id} is not in {list(id_to_entity)}")
```

File: src/ibek/ioc.py [hunk 2]
```diff
@@ -122,14 +124,15 @@ def make_entity_classes(support: Support) -> types.SimpleNamespace:
     setattr(modules, support.module, module)
     modules.__all__.append(support.module)
     for definition in support.definitions:
-        entity_cls = make_entity_class(definition, support)
-        setattr(module, definition.name, entity_cls)
+        id_to_entity = make_entity_class(definition, support)
+        setattr(module, definition.name, id_to_entity)
```
Comment (thomascobb) [inline src/ibek/ioc.py:128]: The old name made more sense

```suggestion
        entity_cls = make_entity_class(definition, support)
        setattr(module, definition.name, entity_cls)
```

File: src/ibek/support.py [hunk 3]
```diff
@@ -13,6 +10,11 @@
 
 from .globals import T, desc
 
+"""
+The Support Class represents a deserialized <MODULE_NAME>.ibek.yaml file.
+It contains a hierarchy of Entity dataclasses.
+"""
+
```
Comment (thomascobb) [inline src/ibek/support.py:17]: I think this should go back up at the top

File: .vscode/settings.json [hunk 4]
```diff
@@ -14,11 +14,11 @@
         "source.organizeImports": true
     },
     "yaml.schemas": {
```
Comment (thomascobb) [inline .vscode/settings.json:16]: I think we can remove "yaml.schemas" from settings.json, and put the "# yaml-language-server:" line at the start of each yaml file instead

File: src/ibek/ioc.py (hunk 2 above)
Comment (coretl) [reply src/ibek/ioc.py]: just here, like you've done

**Review by coretl — APPROVED** [review]

**Pull request merged**